File: elixir/bob/bob.exs
defmodule Bob do

  def hey(input) do
    cond do
      silence?(input) ->
        "Fine. Be that way!"
      yell?(input) and question?(input) ->
        "Calm down, I know what I'm doing!"
      question?(input) ->
        "Sure."
      yell?(input) ->
        "Whoa, chill out!"
      true ->
        "Whatever."
    end
  end

  defp yell?(input) do String.upcase(input) == input and String.match?(input, ~r/\p{L}+/) end
  defp silence?(input) do String.trim(input) == "" end
  defp question?(input) do String.ends_with?(input, "?") end
end

File: python/bob/bob.py
def hey(phrase):
	# clean any whitespaces 
	phrase = phrase.strip()

	if not phrase:
		return 'Fine. Be that way!'
	
	is_upper = phrase.isupper()
	has_question_mark = phrase.endswith("?")

	if is_upper and has_question_mark:
		return 'Calm down, I know what I\'m doing!'

	if has_question_mark:
		return 'Sure.'

	if is_upper:
		return 'Whoa, chill out!'

	return 'Whatever.'
